engines/myst3/state.h

```diff
@@ -48,12 +49,13 @@ enum ViewType {
 
 class GameState {
 public:
-	GameState(const Common::Platform platform, Database *database);
+	GameState(const Common::Platform platform, Database *database, Myst3Engine *vm);
```

Comment:
Game state intentionally did not have a dependency onto `Myst3Engine` for future plans I have with that code. Please  move the part of the code that opens the file to `Myst3Engine`.